lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'active_support'
+require 'active_support/core_ext/object/deep_dup'
+
+# AutoHCK module
+module AutoHCK
+  # UtilsParser class
+  class UtilsParser
+    # Converts string and float input to int. Strips all non-numeric
+    # characters from strings.
+    #
+    # @param value [String, Float] Input value
+    #
+    # @return [Integer, nil] Integer if successful conversion, otherwise nil
+    def self.convert_to_int(value)
+      if value.is_a?(String)
+        str_val = value.gsub(/[^0-9\-.]/, '')
+        begin
+          Integer(str_val)
+        rescue ArgumentError, TypeError
+          begin
+            Integer(Float(str_val))
+          rescue ArgumentError, TypeError
+            nil
+          end
+        end
+      elsif value.is_a?(Numeric)
+        value.to_i
+      end
+    end
+  end
+
+  # SysInfoParser class
+  class SysInfoParser
+    DELIM = ':' # k/v delimiter
+
+    # Main text parsing function
+    #
+    # @param data [String] Text data to parse
+    #
+    # @return [Array<Hash>] List of dictionaries. Processed structured data.
+    def parse(data)
+      rows = CSV.parse(data, headers: true)
+      raw_data = rows.first.to_h
+
+      raw_output = {}
+      raw_data.each do |k, v|
+        raw_output[k] = case k
+                        when 'Hotfix(s)', 'Processor(s)'
+                          parse_hotfixs_or_processors(v)
+                        else
+                          v.strip
+                        end
+      end
+
+      raw_output
+    end
+
+    private
+
+    # Turns a list of return-delimited hotfixes or processors
+    # with [x] as a prefix into an array of hotfixes
+    #
+    # Note that if a high number of items exist, the list may cutoff
+    # and this list may not contain all items installed on the system
+    #
+    # IRL, this likely applies to hotfixes more than processors
+    #
+    # Parameters:
+    #   data: (string) Input string
+    def parse_hotfixs_or_processors(data)
+      arr_output = []
+      data.split(',').each_with_index do |l, i|
+        # skip line that says how many are installed
+        next if i.zero?
+
+        # we have to make sure this is a complete line
+        # as data could cutoff. Make sure the delimiter
+        # exists. Otherwise, skip it.
+        next unless l.include?(DELIM)
```

Comment:
Can that cut off really happen, especially when it's told to output a machine-readable CSV?

Reply:
droped